lib/logput/adapters/base.rb

```diff
@@ -0,0 +1,34 @@
+# Logput
+module Logput
+  # Logging Adapters
+  module Adapters
+    # Base class from which all adapters inherit
+    class Base
+      # Initialize
+      def initialize(logger)
+        @logger = logger
+      end
+
+      # Registers a new adapter
+      #
+      # @param [Symbol] adapter The name of the adapter
+      def self.register(adapter)
+        raise "Already Registered :#{adapter}" if Logput::Adapters.registered_adapters[adapter]
+        Logput::Adapters.registered_adapters[adapter] = self
+      end
+
+      # Placeholder for handles? method to be overridden when subclassed
+      # @param [Class] logger
+      # @return [Boolean]
+      def self.handles?(logger)
```

Comment:
Even though this is a placeholder method intended to ensure subclasses provide this method, Rubocop has a suggestion:

```
Unused method argument - logger. If it's necessary, use _ or _logger as an argument name to indicate that it won't be used. You can also write as handles?(*) if you want the method to accept any arguments but don't care about them.
```

So i'll update to use `_logger` as the param name here.